### PR TITLE
feat: support iron useful tracepoints

### PR DESCRIPTION
**Pull request by ymski**
## Description

This PR will implement the following contents:
- Record newly added trace points
- Filter for newly added trace points

The trace points added in iron are the following:

- ros2:ring_buffer_construct
- ros2:ring_buffer_enqueue
- ros2:ring_buffer_dequeue
- ros2:ring_buffer_clear
- ros2:buffer_to_ipb
- ros2:ipb_to_subscription

In addition, the following existing tracepoints have been modified for addition to rolling

- ros2_caret:caret_init

## Related links

- https://github.com/tier4/CARET_analyze/pull/318

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: CARET_trace/src/ros_trace_points.cpp [hunk 1]
```diff
@@ -1068,5 +1116,181 @@ void ros_trace_rmw_subscription_init(
...
+  }
+}
+
+void ros_trace_rclcpp_ring_buffer_clear(
+    const void * buffer
+)
+{
+  static void * orig_func = dlsym(RTLD_NEXT, __func__);
+  using functionT = void (*)(const void *);
+  ((functionT) orig_func)(buffer);
```
Comment (isp-uetsuki) [inline CARET_trace/src/ros_trace_points.cpp:1287]: Is checking by `controller.is_allowed_buffer` and `context.is_recording_allowed` not necessary for this tracepoint?

**Review by isp-uetsuki — CHANGES_REQUESTED** [review]

File: CARET_trace/src/ros_trace_points.cpp (hunk 1 above)
Comment (ymski) [reply CARET_trace/src/ros_trace_points.cpp]: fixed in [64a6050](https://github.com/tier4/CARET_trace/pull/150/commits/64a60504e0a91e4b28f5535f92f65f5f5422444f).

File: CARET_trace/include/caret_trace/tracing_controller.hpp [hunk 2]
```diff
@@ -79,6 +79,16 @@ class TracingController
...
 
+  /// @brief Register topic name for ros_trace_rcl_publisher_init
+  /// @param buffer  Address of the buffer.
+  /// @param ipb  Address of the IntraProcessBuffer.
+  void add_buffer(const void * buffer, const void * ipb);
+
+  /// @brief Register topic name for ros_trace_rcl_publisher_init
+  /// @param ipb  Address of the IntraProcessBuffer.
+  /// @param subscription  Address of the subscription instance.
+  void add_ipb(const void * ipb, const void * subscription);
```
Comment (isp-uetsuki) [inline CARET_trace/include/caret_trace/tracing_controller.hpp:90]: Please modify `@brief` comment to fit with function of these method.

**Review by isp-uetsuki — CHANGES_REQUESTED** [review]

**Review by isp-uetsuki — APPROVED** [review]
LGTM

**Review by nabetetsu — APPROVED** [review]
LGTM :+1:

**Pull request merged**